src/Heap.sol

```diff
@@ -36,209 +40,202 @@ library BasicHeap {
     /// @param _heap The heap to modify.
     /// @param _id The address of the account to insert.
     /// @param _value The value of the account to insert.
-    function insert(
-        Heap storage _heap,
-        address _id,
-        uint256 _value
-    ) internal {
-        if (_value == 0) revert WrongValue();
+    function insert(Heap storage _heap, address _id, uint256 _value) internal {
         if (_id == address(0)) revert AddressIsZero();
-        if (_heap.ranks[_id] != 0) revert AccountAlreadyInserted();
 
-        // Put the account at the end of the heap.
-        _heap.accounts.push(Account(_id, _value));
-        uint256 size = getSize(_heap);
-        _heap.ranks[_id] = size;
+        uint256 accountsLength = _heap.accounts.length;
+        if (_containsAccount(_heap, _heap.indexOf[_id], accountsLength, _id)) revert AccountAlreadyInserted();
```

Comment:
In heap orderings, the check is implicitely done [here](https://github.com/morpho-org/morpho-data-structures/blob/47000974e8ef9cce097b86d2326de33f19320ad1/src/HeapOrdering.sol#L50-L56). Because the only entry point is the function `update`, nothing needs to be checked after.